cpp/src/sort/sort.cuh

```diff
@@ -0,0 +1,70 @@
+// -*-c++-*-
+
+#pragma once
+
+#include <cudf/types.h>
+
+namespace cusort {
```

Comment:
If we are initially placing these functions under cugraph, shouldn't we put this in the cugraph namespace?

Reply:
I imagined that putting them in the cusort namespace would reduce the number of code changes later when we move it, but I'm find putting it in either cugraph or cusort.

Reply:
Depends on the intent. In cugraph only internal functions are in cugraph namespace. The public API is not encapsulated in a namespace. It is a C API legacy thing. Adding it under cugraph namespace really indicates it is a cugraph internal function.

Reply:
OK, if we are planning to move to the C++ API, and anything here is temporary, I have no complaint.

Reply:
Since the long-term plan is to move sorting into RAFT (ml-graph primitives repo) the namespace might need to change.  As Alex's pointed out, since it is not going to be a cugrapn internal function, maybe 'cugraph' is not the right namespace.  I'm okay leaving it as 'cusort' for now.